plugin/pulpcore/plugin/models/importer.py

```diff
@@ -25,7 +25,7 @@ class Importer(PlatformImporter):
     :class: `pulpcore.plugin.serializers.repository.ImporterSerializer`.
     """
 
-    def sync(self):
+    def sync(self, new_version, old_version):
```

Comment:
What about just passing in new_version and leaving out old_version? The plugin code can easily query for other versions as it needs them. Most cases I believe will only require new_version. Also what about calling it `repo_version` instead?

Reply:
``old_version`` is how a plugin will determine what content is currently in the repo, which is required information when trying to decide what to add and remove. Given that, every sync operation will require both of these.

Reply:
As soon as new_version is created, it will implicitly have all of the same content. Couldn't the plugin writer look at new_version and answer the same question?

Reply:
new_version may change as the sync does what it does, so it's not a reliable source of information about what was in the previous version. The previous version is explicitly the immutable content set that the plugin needs in order to create a new content set. So I think the simplest and least-confusing thing to do is give the plugin a direct reference to the content set it is interested in.

Reply:
My suggestion is to adopt this for now ([since it works and users can sync pulp_file repos](https://github.com/pulp/pulp_file/pull/20)) and then meet up to develop a comprehensive design around the plugin api for repo versions when we address the plugin writer user stories. Thoughts?

Reply:
That sounds good to me. It would be a big step forward. We can fix the other parts based on the user stories.